### ISSUE TITLE
SPDX 2.3 Support

### ISSUE THREAD

**Issue by lumjjb** [spdx/tools-golang#156]
Implement SPDX 2.3 support (preferably waiting on https://github.com/spdx/tools-golang/issues/152) for refactor changes. 

2.3 Link: https://spdx.github.io/spdx-spec/v2.3/

**Comment by nishakm** [comment]
Need help with this? I need a 2.3 model to implement a generator :)

**Comment by lumjjb** [comment]
that would be super helpful!!!!

**Comment by kzantow** [comment]
@nishakm I also need some of the new features of 2.3 -- could I help with this?

**Comment by nishakm** [comment]
> @nishakm I also need some of the new features of 2.3 -- could I help with this?

Absolutely!

**Comment by lumjjb** [comment]
Assigned the issue to bot of you!!! Thank you!!

**Comment by kzantow** [comment]
I'm starting to look at implementing this -- is there a document somewhere with the changes for 2.3?

**Comment by lumjjb** [comment]
Awesome @kzantow !!!! Yes! Here's the changes from 2.2 (https://spdx.github.io/spdx-spec/v2.3/diffs-from-previous-editions/)! Is this something that's useful?

You can ignore the between v2.2.x versions because they are mainly documentation changes.

**Comment by kzantow** [comment]
@lumjjb I've [added a draft PR](https://github.com/spdx/tools-golang/pull/164) with some initial changes for 2.3: namely a new v2_3 data model and JSON support, along with a number of other changes. It does not yet have the other format support (it will before I'd consider it done).

It looks like you did some of the refactoring for the current state of this repo, if you had a little time, could you have a quick look and make sure I'm on the right track before I complete everything else?

**Comment by kzantow** [comment]
@lumjjb I believe I've implemented all the 2.3 support in this PR: https://github.com/spdx/tools-golang/pull/164

**Comment by kzantow** [comment]
I've also added a refactoring suggestion I would like to do here: https://github.com/spdx/tools-golang/issues/165 I know there are some other things (like translating between versions) that people would like to see, but I feel that's a separate issue and the refactoring would dramatically improve the experience every time someone needs to implement a new SPDX version.
